lib/driver.rb

```diff
@@ -17,31 +18,54 @@ def initialize(input)
       @name = input[:name]
       @vehicle_id = input[:vin]
       @status = input[:status] == nil ? :AVAILABLE : input[:status]
-
       @trips = input[:trips] == nil ? [] : input[:trips]
     end
 
+    def finished_trips
+      trips.reject {|trip| trip.end_time == nil}
+    end
```

Comment:
I love this helper method! Good job breaking the problem up into multiple smaller pieces.

Reply:
However, it seems like this should call `trip.finished?` instead of manually checking `trip.end_time == nil`. 

Bringing it back to the Metz reading, calling `trip.finished?` is an example of relying on _behavior_, whereas checking `trip.end_time` is relying on _data_.